tests/test_core.py

```diff
@@ -107,6 +107,12 @@ def test_read_modules_from_input_should_set_modules_attr_when_called(simple_sim)
     assert simple_sim.physics_modules[0].owner == simple_sim
     assert simple_sim.physics_modules[0].input_data == {"name": "ExampleModule"}
 
+def test_find_tool_by_name_should_identify_one_tool(simple_sim):
```

Comment:
This test would be better if there are two tools in the config of the same type, but with different custom names.